python/lsst/daf/butler/core/utils.py

```diff
@@ -103,7 +103,8 @@ def globToRegex(
     if not expressions or "*" in expressions:
         return Ellipsis
 
-    nomagic = re.compile(r"^[\w/\.\-@]+$", re.ASCII)
+    # The character class range at the end adds support for emoji.
+    nomagic = re.compile(r"^[\w/\.\-@\u263a-\U0001f645]+$")
```

Comment:
This is a little brute force in terms of character ranges. There are more limited ranges that can be supported for emoticons and flags etc rather than allowing all emoji.

Reply:
Seems to me that you should give up on trying to identify legal characters and instead merely detect whether there are glob characters that need `fnmatch`:
```
nomagic = re.compile(r"^[^*?[\]]$")
```

Reply:
Okay. That is probably better. This is specifically designed for `fnmatch` -- I wasn't entirely sure of the full list but I'll assume the above is correct.

Reply:
I took the list from https://docs.python.org/3/library/fnmatch.html

Reply:
In fact `fnmatch` doesn't seem to treat a `]` as special unless it's paired with `[`, so you could leave the `\]` out.

Reply:
Oh, and you could invert the logic then, making it just an RE to look for one of the three characters (`*?[`) anywhere in the string.

Reply:
Yes, I'm inverting.